src/asm_files.cpp

```
@@ -309,344 +379,639 @@ static elf_global_data parse_map_sections(const parse_params_t& parse_params, co
+    }
+}
 
-        // Perform function relocations and fill in the inst.imm values of CallLocal instructions.
-        std::map<std::string, ELFIO::Elf_Xword> subprogram_offsets;
-        for (const auto& reloc : function_relocations) {
-            if (reloc.prog_index >= raw_programs.size()) {
-                continue;
-            }
-            if (raw_programs[reloc.prog_index].function_name != prog.function_name) {
+void ProgramReader::process_core_relocations(const libbtf::btf_type_data& btf_data) {
+    const ELFIO::section* relo_sec = reader.sections[".rel.BTF"];
```

Clang (14.0.6)

[warning] variable 'relo_sec' is not initialized

(cppcoreguidelines-init-variables)